staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -28,38 +31,31 @@ import (
 // The key type is unexported to prevent collisions
 type key int
 
-const (
-	// auditAnnotationsKey is the context key for the audit annotations.
-	// TODO: consolidate all audit info under the AuditContext, rather than storing 3 separate keys.
-	auditAnnotationsKey key = iota
+// auditKey is the context key for storing the audit context that is being
+// captured and the evaluated policy that applies to the given request.
+const auditKey key = iota
 
-	// auditKey is the context key for storing the audit event that is being
-	// captured and the evaluated policy that applies to the given request.
-	auditKey
+// AuditContext holds the information for constructing the audit events for the current request.
+type AuditContext struct {
```

Comment:
It was much worse before, but it's still pretty weird that some of this is private and accessed via Get/With functions on the context. Do you plan to clean it up more later?

Reply:
Yes. I started the cleanup here, but the PR got huge, so I broke this piece out. My plan is to make all this private, and expose varies `Record*` functions to add info to the audit event, and expose the bare minimum through getters.

Reply:
OK, I don't think I object if SIG Auth is on board.